mlir/lib/Dialect/Tosa/Transforms/TosaValidation.cpp

```diff
@@ -526,8 +527,8 @@ void TosaValidation::runOnOperation() {
     if (failed(applyLevelCheck(op)))
       signalPassFailure();
 
-    // do variable type checks
-    if (failed(applyVariableCheck(op)))
+    // do variable type checks iff topOp is a ModuleOp
+    if (isModule && failed(applyVariableCheck(op)))
```

Comment:
That makes the validation behavior different depending on how it is scheduled: seems like something easy to misconfigure? What is the motivation to allow this "partial validation"?